src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -104,34 +106,50 @@ public void put(Collection<SinkRecord> records) throws ConnectException {
     partitionPauser.maybeResumePartitions();
 
     for (SinkRecord record : records) {
+      verifyChangingTopic(record);
       OffsetState offsetState = offsetTracker.addPendingRecord(record);
-
       if (shouldSkipRecord(record)) {
         logTrace("Ignoring {} with null value.", record);
         offsetState.markProcessed();
         reportBadRecord(record, new ConnectException("Cannot write null valued record."));
         continue;
       }
-
       logTrace("Writing {} to Elasticsearch.", record);
-
       tryWriteRecord(record, offsetState);
     }
     partitionPauser.maybePausePartitions();
   }
 
+  /**
+   * Fail fast in cases where topic mutating SMTs are used and connector configuration doesn't
+   * support it.
+   */
+  private void verifyChangingTopic(SinkRecord record) {
```

Comment:
should this method also be moved to OffsetTracker interface?

Reply:
Great point, `addPendingRecord` looks like a perfect fit for this logic.